src/ansible_navigator/configuration_subsystem/navigator_post_processor.py

```diff
@@ -46,13 +48,39 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+@dataclass
```

Comment:
(educational) Not intended to suggest one or the other, I am curious why not a namedtuple sine it shouldn't change....

Reply:
@cidrblock dataclasses implement a far more superior and standardized concept, and they also can be frozen (and should be most of the time — immutable objects are easier to reason about, they are safer). Also, AFAIR dataclasses use `__slots__` as an optimization.